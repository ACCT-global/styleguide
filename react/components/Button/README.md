Types

```js
<div>
  <div>
    <span className="mr4">
      <Button variation="primary">Primary</Button>
    </span>
    <span className="mr4">
      <Button variation="secondary">Secondary</Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary">Tertiary</Button>
    </span>
  </div>
  <div className="mt4">
    <span className="mr4">
      <Button variation="primary" disabled>Primary</Button>
    </span>
    <span className="mr4">
      <Button variation="secondary" disabled>Secondary</Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary" disabled>Tertiary</Button>
    </span>
  </div>
</div>
```

Sizes

```js
<div>
  <div className="mb4">
    <span className="mr4">
      <Button variation="primary" size="small">
        Small
      </Button>
    </span>
    <span className="mr4">
      <Button variation="primary">
        Default
      </Button>
    </span>
    <span className="mr4">
      <Button variation="primary" size="large">
        Large
      </Button>
    </span>
  </div>
  <div className="mb4">
    <span className="mr4">
      <Button variation="secondary" size="small">
        Small
      </Button>
    </span>
    <span className="mr4">
      <Button variation="secondary" size="regular">
        Default
      </Button>
    </span>
    <span className="mr4">
      <Button variation="secondary" size="large">
        Large
      </Button>
    </span>
  </div>
  <div className="mb4">
    <span className="mr4">
      <Button variation="tertiary" size="small">
        Small
      </Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary" size="regular">
        Default
      </Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary" size="large">
        Large
      </Button>
    </span>
  </div>
</div>
```

Box types

```js
<div>
  <div className="mb4">
    <Button variation="primary">Default</Button>
  </div>
  <div className="mb4">
    <Button variation="primary" block>
      Block
    </Button>
  </div>
</div>
```

Colored container background

```js
<div className="bg-washed-blue pa5">
  <div>
    <span className="mr4">
      <Button variation="primary">Primary</Button>
    </span>
    <span className="mr4">
      <Button variation="secondary">Secondary</Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary">Neutral</Button>
    </span>
  </div>
  <div className="mt4">
    <span className="mr4">
      <Button variation="primary" disabled>Primary</Button>
    </span>
    <span className="mr4">
      <Button variation="secondary" disabled>Secondary</Button>
    </span>
    <span className="mr4">
      <Button variation="tertiary" disabled>Neutral</Button>
    </span>
  </div>
</div>
```

With icon

```js
const CloseIcon = require('../icon/Close').default
;<Button icon variation="primary">
  <CloseIcon color="#fff" />
</Button>
```

Loading state

```js
initialState = { isLoading1: false, isLoading2: true };
<div>
  <div>
    <Button
      variation="primary"
      onClick={() => setState({ isLoading1: !state.isLoading1 })}
      isLoading={state.isLoading1}
    >
      Toggle loading state
    </Button>
  </div>
  <div className="mt4">
    <Button
      variation="primary"
      onClick={() => setState({ isLoading2: !state.isLoading2 })}
      isLoading={state.isLoading2}
    >
      Toggle loading state
    </Button>
  </div>
</div>
```
