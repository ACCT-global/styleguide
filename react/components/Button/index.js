import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'

import Spinner from '../Spinner'

class Button extends Component {
  handleClick = event => {
    !this.props.disabled &&
      !this.props.isLoading &&
      this.props.onClick &&
      this.props.onClick(event)
  }

  render() {
    const {
      size,
      block,
      variation,
      icon,
      children,
      isLoading,
      collapseLeft,
      collapseRight,
    } = this.props

    const disabled = this.props.disabled || isLoading

    let classes = 'vtex-button bw1 ba fw5 br2 v-mid relative pa0 '
    let loaderSize = 15
    let horizontalPadding = 0

    switch (size) {
      case 'small':
        classes += 'h-small t-action--small '
        horizontalPadding = icon ? 3 : 5
        break
      default:
      case 'regular':
        classes += 'h-regular t-action '
        horizontalPadding = icon ? 4 : 6
        classes += 't-action '
        loaderSize = 20
        break
      case 'large':
        classes += `h-large t-action--large `
        horizontalPadding = icon ? 5 : 7
        loaderSize = 25
        break
    }

    classes += `ph${horizontalPadding} `

    if (collapseLeft) {
      classes += `nl${horizontalPadding} `
    }
    if (collapseRight) {
      classes += `nr${horizontalPadding} `
    }
    if (icon) {
      classes += 'icon-button dib '
    }

    switch (variation) {
      default:
      case 'primary': {
        if (disabled) {
          classes += 'bg-disabled b--disabled c-on-disabled '
        } else {
          classes +=
            'bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary '
        }
        break
      }
      case 'secondary': {
        if (disabled) {
          classes += 'bg-disabled b--disabled c-on-disabled '
        } else {
          classes +=
            'bg-action-secondary b--action-secondary c-on-action-secondary hover-bg-action-secondary hover-b--action-secondary hover-c-on-action-secondary '
        }
        break
      }
      case 'tertiary': {
        if (disabled) {
          classes += 'bg-transparent b--transparent c-disabled '
        } else {
          classes +=
            'bg-transparent b--transparent c-action-primary hover-b--transparent hover-c-action-primary '
        }
        break
      }
      case 'danger': {
        if (disabled) {
          classes += 'bg-disabled b--disabled c-on-disabled '
        } else {
          classes +=
            'bg-danger b--danger c-on-danger hover-bg-danger hover-b--danger hover-c-on-danger '
        }
        break
      }
      case 'danger-tertiary': {
        if (disabled) {
          classes += 'bg-transparent b--transparent c-disabled '
        } else {
          classes +=
            'bg-transparent b--transparent c-danger hover-b--transparent hover-c-danger '
        }
        break
      }
    }

    if (!disabled) {
      classes += 'pointer '
    }

    if (block) {
      classes += 'w-100 '
    }

    return (
      <button
        id={this.props.id}
        autoFocus={icon ? undefined : this.props.autoFocus}
        disabled={icon ? undefined : this.props.disabled}
        name={icon ? undefined : this.props.name}
        type={icon ? undefined : this.props.type}
        value={icon ? undefined : this.props.value}
        tabIndex={0}
        className={classes}
        onClick={this.handleClick}
        onMouseEnter={this.props.onMouseEnter}
        onMouseLeave={this.props.onMouseLeave}
        onMouseOver={this.props.onMouseOver}
        onMouseOut={this.props.onMouseOut}
        onMouseUp={this.props.onMouseUp}
        onMouseDown={this.props.onMouseDown}
        ref={this.props.ref}
        style={icon ? { fontSize: 0 } : {}}>
        {isLoading ? (
          <Fragment>
            <span className="left-0 w-100 absolute flex justify-center items-baseline">
              <Spinner
                secondary={variation === 'primary' || variation === 'danger'}
                size={loaderSize}
              />
            </span>
            <span style={{ opacity: 0 }}>{children}</span>
          </Fragment>
        ) : (
          <div className="flex items-center justify-center h-100">
            {children}
          </div>
        )}
      </button>
    )
  }
}

Button.defaultProps = {
  size: 'regular',
  block: false,
  variation: 'primary',
  disabled: false,
  autoFocus: false,
  icon: false,
  type: 'button',
  isLoading: false,
}

Button.propTypes = {
  /** Button size  */
  size: PropTypes.oneOf(['small', 'regular', 'large']),
  /** Button prominence variation */
  variation: PropTypes.oneOf([
    'primary',
    'secondary',
    'tertiary',
    'danger',
    'danger-tertiary',
  ]),
  /** Block style */
  block: PropTypes.bool,
  /** Loading state */
  isLoading: PropTypes.bool,
  /** If you are using just an Icon component inside, use this as true */
  icon: PropTypes.bool,
  /** (Button spec attribute) */
  id: PropTypes.string,
  /** (Button spec attribute) */
  autoFocus: PropTypes.bool,
  /** (Button spec attribute) */
  autoComplete: PropTypes.string,
  /** (Button spec attribute) */
  disabled: PropTypes.bool,
  /** (Button spec attribute) */
  name: PropTypes.string,
  /** (Button spec attribute) */
  type: PropTypes.string,
  /** (Button spec attribute) */
  value: PropTypes.string,
  /** Label of the Button */
  children: PropTypes.node.isRequired,
  /** onClick event */
  onClick: PropTypes.func,
  /** onMouseEnter event */
  onMouseEnter: PropTypes.func,
  /** onMouseLeave event */
  onMouseLeave: PropTypes.func,
  /** onMouseOver event */
  onMouseOver: PropTypes.func,
  /** onMouseOut event */
  onMouseOut: PropTypes.func,
  /** onMouseUp event */
  onMouseUp: PropTypes.func,
  /** onMouseDown event */
  onMouseDown: PropTypes.func,
  /** ref function */
  ref: PropTypes.func,
  /** Cancels out left padding */
  collapseLeft: PropTypes.bool,
  /** Cancels out right padding */
  collapseRight: PropTypes.bool,
}

export default Button
