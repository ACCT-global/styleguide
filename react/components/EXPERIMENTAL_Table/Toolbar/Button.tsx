import React, { forwardRef, ReactNode } from 'react'
import csx from 'classnames'

import ButtonWithIcon from '../../ButtonWithIcon/index.js'
import { BUTTON } from '../constants'
const ICON_OPTICAL_COMPENSATION = { marginTop: 1.5 }

type Ref = HTMLDivElement

const Button = forwardRef<Ref, ButtonProps>(
  (
    {
      id,
      title,
      disabled,
      onClick,
      icon,
      isLoading,
      children,
      label,
      variation,
      isActiveOfGroup,
      isGrouped,
      isFirstOfGroup,
      size,
    },
    ref
  ) => {
    const isTertiary = variation === BUTTON.VARIATION.TERTIARY
    return (
      <div
        id={id}
        title={title}
        ref={ref}
        className={csx('relative', { mh2: isTertiary })}>
        <ButtonWithIcon
          icon={
            <span
              className={`${isTertiary ? 'c-on-base mh2' : ''}`}
              style={ICON_OPTICAL_COMPENSATION}>
              {icon}
            </span>
          }
          isGrouped={isGrouped}
          isFirstOfGroup={isFirstOfGroup}
          isActiveOfGroup={isActiveOfGroup}
          disabled={disabled}
          isLoading={isLoading}
          variation={variation}
          size={size}
          onClick={onClick}>
          {label && (
            <span className={isTertiary ? 'c-on-base' : ''}>{label}</span>
          )}
        </ButtonWithIcon>
        {children}
      </div>
    )
  }
)

export type ButtonProps = {
  id?: string
  label?: string
  onClick?: Function
  isLoading?: boolean
  disabled?: boolean
  size?: Size
  icon?: any
  title?: string
  variation?: Variation
  isActiveOfGroup?: boolean
  isGrouped?: boolean
  isFirstOfGroup?: boolean
  children?: ReactNode
}

Button.defaultProps = {
  variation: BUTTON.VARIATION.TERTIARY,
  isActiveOfGroup: false,
  isGrouped: false,
  isFirstOfGroup: false,
  size: BUTTON.SIZE.SMALL,
}

export default Button
