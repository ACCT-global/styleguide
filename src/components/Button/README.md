Default

```js
<Button>Save</Button>
```

Secondary

```js
<Button secondary>Save</Button>
```

Primary

```js
<Button primary>Save</Button>
```

Disabled

```js
<Button disabled>Save</Button>
```

Side by side

```js
<div>
  <Button>Save</Button>
  <Button secondary>Save</Button>
  <Button primary>Save</Button>
  <Button disabled>Save</Button>
</div>
```

With spinner

```js
<div>
  <Button isLoading>Save</Button>
  <Button primary isLoading>
    Save
  </Button>
  <Button secondary isLoading>
    Save
  </Button>
  <Button disabled isLoading>
    Save
  </Button>
</div>
```

With icon

```js
<table height="50" width="50">
  <tbody>
    <tr>
      <td>
        <Button primary>
          <Icon type="check" fill="fff" />
        </Button>
      </td>
    </tr>
  </tbody>
</table>
```
