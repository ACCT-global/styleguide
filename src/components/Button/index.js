import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Button extends Component {
  handleClick = event => {
    !this.props.disabled && this.props.onClick && this.props.onClick(event)
  };

  render() {
    const {
      large,
      xLarge,
      block,
      primary,
      secondary,
      disabled,
      icon,
    } = this.props
    const Tag = icon ? 'div' : 'button'

    if (secondary && primary) {
      throw new Error('Button component cannot be primary AND secondary')
    }

    if (large && xLarge) {
      throw new Error(
        'Button component cannot have two sizes at the same time'
      )
    }

    let classes = 'bw1 ba fw5 ttu br2 fw4 '

    classes += icon ? 'icon-button dib ' : ''

    if (large) {
      classes += icon ? 'pa4 ' : 'pv4 ph6 '
      classes += 'f5 '
    } else if (xLarge) {
      classes += icon ? 'pa5 ' : 'pv5 ph7 '
      classes += 'f4 '
    } else {
      classes += icon ? 'pa3 ' : 'pv3 ph5 '
      classes += 'f6 '
    }

    if (!secondary && !primary && !disabled) {
      classes += 'b--transparent blue bg-transparent hover-heavy-blue hover-b--transparent '
    }

    if (secondary && !disabled) {
      classes += 'bg-white b--blue blue hover-white '
    }
    if (secondary) {
      classes += 'hover-bg-blue '
    }

    if (primary && !disabled) {
      classes += 'b--blue bg-blue white hover-bg-heavy-blue hover-b--heavy-blue '
    }

    if (disabled) {
      classes += 'b--light-gray bg-light-gray gray '
    } else {
      classes += 'pointer '
    }

    if (block) {
      classes += 'w-100 '
    }

    return (
      <Tag
        id={this.props.id}
        autoFocus={icon ? undefined : this.props.autoFocus}
        disabled={icon ? undefined : this.props.disabled}
        name={icon ? undefined : this.props.name}
        type={icon ? undefined : this.props.type}
        value={icon ? undefined : this.props.value}
        className={classes}
        onClick={this.handleClick}
        onMouseEnter={this.props.onMouseEnter}
        onMouseLeave={this.props.onMouseLeave}
        onMouseOver={this.props.onMouseOver}
        onMouseOut={this.props.onMouseOut}
        onMouseUp={this.props.onMouseUp}
        onMouseDown={this.props.onMouseDown}
        ref={this.props.ref}
      >
        {this.props.children}
      </Tag>
    )
  }
}

Button.defaultProps = {
  large: false,
  xLarge: false,
  block: false,
  primary: false,
  secondary: false,
  disabled: false,
  autoFocus: false,
  icon: false,
  type: 'button',
}

Button.propTypes = {
  /** Large style */
  large: PropTypes.bool,
  /** xLarge style */
  xLarge: PropTypes.bool,
  /** Block style */
  block: PropTypes.bool,
  /** Primary style */
  primary: PropTypes.bool,
  /** Secondary style */
  secondary: PropTypes.bool,
  /** If you are using just an Icon component inside, use this as true */
  icon: PropTypes.bool,
  /** (Button spec attribute) */
  id: PropTypes.string,
  /** (Button spec attribute) */
  autoFocus: PropTypes.bool,
  /** (Button spec attribute) */
  autoComplete: PropTypes.string,
  /** (Button spec attribute) */
  disabled: PropTypes.bool,
  /** (Button spec attribute) */
  name: PropTypes.string,
  /** (Button spec attribute) */
  type: PropTypes.string,
  /** (Button spec attribute) */
  value: PropTypes.string,
  /** Label of the Button */
  children: PropTypes.node.isRequired,
  /** onClick event */
  onClick: PropTypes.func,
  /** onMouseEnter event */
  onMouseEnter: PropTypes.func,
  /** onMouseLeave event */
  onMouseLeave: PropTypes.func,
  /** onMouseOver event */
  onMouseOver: PropTypes.func,
  /** onMouseOut event */
  onMouseOut: PropTypes.func,
  /** onMouseUp event */
  onMouseUp: PropTypes.func,
  /** onMouseDown event */
  onMouseDown: PropTypes.func,
  /** ref function */
  ref: PropTypes.func,
}

export default Button
