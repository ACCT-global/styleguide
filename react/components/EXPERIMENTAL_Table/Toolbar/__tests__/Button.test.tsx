import React from 'react'
import { render } from 'react-testing-library'

import Button, { ButtonSize, ButtonVariation } from '../Button'

describe('Table V2 Toolbar <Button /> spec', () => {
  const basicProps = {
    id: 'id',
    testId: 'testId',
    label: 'label',
    title: 'title',
    onClick: () => {},
  }
  it('renders with and without icon', () => {
    const icon = <div>Icon</div>
    const button = render(
      <>
        <Button {...basicProps}>basic</Button>
        <Button {...basicProps} icon={icon}>
          withIcon
        </Button>
      </>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
  it('renders with different states', () => {
    const button = render(
      <>
        <Button {...basicProps} isLoading>
          loading
        </Button>
        <Button {...basicProps} disabled>
          disabled
        </Button>
      </>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
  it('renders all the sizes', () => {
    const button = render(
      <>
        <Button {...basicProps} size={ButtonSize.Small}>
          Testing small
        </Button>
        <Button {...basicProps} size={ButtonSize.Regular}>
          Testing regular
        </Button>
        <Button {...basicProps} size={ButtonSize.Large}>
          Testing large
        </Button>
      </>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
  it('renders as group', () => {
    const basicProps = {
      id: 'id',
      testId: 'testId',
      label: 'label',
      onClick: () => {},
    }
    const button = render(
      <Button {...basicProps} isActiveOfGroup isGrouped isFirstOfGroup>
        Testing
      </Button>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
  it('renders all variations', () => {
    const button = render(
      <>
        <Button {...basicProps} variation={ButtonVariation.Primary}>
          Testing Primary
        </Button>
        <Button {...basicProps} variation={ButtonVariation.Secondary}>
          Testing Secondary
        </Button>
        <Button {...basicProps} variation={ButtonVariation.Tertiary}>
          Testing Tertiary
        </Button>
      </>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
})
