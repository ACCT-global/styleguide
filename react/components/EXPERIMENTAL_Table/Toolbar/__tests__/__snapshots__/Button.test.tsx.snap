// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 Toolbar <Button /> spec renders all the sizes 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing small
  </div>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-regular t-action bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph6 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr2"
        >
          <div
            class="mr3 nl3 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing regular
  </div>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-large t-action--large bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph7 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr3"
        >
          <div
            class="mr4 nl4 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing large
  </div>
</DocumentFragment>
`;

exports[`Table V2 Toolbar <Button /> spec renders all variations 1`] = `
<DocumentFragment>
  <div
    class="relative"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class=""
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class=""
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing Primary
  </div>
  <div
    class="relative"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-secondary b--action-secondary c-on-action-secondary hover-bg-action-secondary hover-b--action-secondary hover-c-on-action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class=""
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class=""
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing Secondary
  </div>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing Tertiary
  </div>
</DocumentFragment>
`;

exports[`Table V2 Toolbar <Button /> spec renders as group 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 br--left min-h-small t-action--small bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary pointer "
      data-testid="testId"
      style="background-color: rgb(12, 56, 159); border-color: #0c389f;"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    Testing
  </div>
</DocumentFragment>
`;

exports[`Table V2 Toolbar <Button /> spec renders with and without icon 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    basic
  </div>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            >
              <div>
                Icon
              </div>
            </span>
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    withIcon
  </div>
</DocumentFragment>
`;

exports[`Table V2 Toolbar <Button /> spec renders with different states 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-disabled "
      data-testid="testId"
      tabindex="0"
      type="button"
    >
      <span
        class="vtex-button__spinner-container top-0 left-0 w-100 h-100 absolute flex justify-center items-center"
      >
        <svg
          class="vtex__icon-spinner  c-action-primary "
          height="15"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <style>
            
            @keyframes vtex-spinner-rotate {
              from {
                transform-origin: 50% 50%;
                transform: rotate(0deg);
              }
              to {
                transform-origin: 50% 50%;
                transform: rotate(360deg);
              }
            }

            @keyframes vtex-spinner-fill {
              0% {
                stroke-dasharray: 0 0 2 251.32741228718345;
              }
              50% {
                stroke-dasharray: 0 0 188.49555921538757 251.32741228718345;
              }
              100% {
                stroke-dasharray: 0 249.32741228718345 188.49555921538757 251.32741228718345;
              }
            }

            .vtex-spinner_circle {
              animation: vtex-spinner-fill 1.25s infinite cubic-bezier(0.455, 0.030, 0.515, 0.955), vtex-spinner-rotate 0.625s infinite linear;
            }
          
          </style>
          <circle
            class="vtex-spinner_circle"
            cx="50"
            cy="50"
            fill="none"
            r="40"
            stroke="currentColor"
            stroke-dasharray="0 0 188.49555921538757 251.32741228718345"
            stroke-dashoffset="1"
            stroke-linecap="round"
            stroke-width="10"
          />
        </svg>
      </span>
      <span
        class="vtex-button__label flex items-center justify-center h-100 ph5  o-0"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-on-base"
            >
              label
            </span>
          </div>
        </span>
      </span>
    </button>
    loading
  </div>
  <div
    class="relative mh2"
    data-testid="testId__container"
    id="id__container"
    title="title"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-disabled "
      data-testid="testId"
      disabled=""
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-muted-2"
              style="margin-top: 1.5px;"
            />
          </div>
          <div>
            <span
              class="c-muted-2"
            >
              label
            </span>
          </div>
        </span>
      </div>
    </button>
    disabled
  </div>
</DocumentFragment>
`;
